File: source/cpBase/CPGroupBaseClass.vb
'
' documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
'
Namespace Contensive.BaseClasses
    Public MustInherit Class CPGroupBaseClass
        '
        ' documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
        '
        Public MustOverride Sub Add(ByVal GroupName As String, Optional ByVal GroupCaption As String = "")
        Public MustOverride Sub AddUser(ByVal GroupNameIdOrGuid As String)
        Public MustOverride Sub AddUser(ByVal GroupNameIdOrGuid As String, ByVal UserId As Integer)
        Public MustOverride Sub AddUser(ByVal GroupNameIdOrGuid As String, ByVal UserId As Integer, ByVal DateExpires As Date)
        Public MustOverride Sub Delete(ByVal GroupNameIdOrGuid As String)
        Public MustOverride Function GetId(ByVal GroupNameIdOrGuid As String) As Integer
        Public MustOverride Function GetName(ByVal GroupNameIdOrGuid As String) As String
        Public MustOverride Sub RemoveUser(ByVal GroupNameIdOrGuid As String, Optional ByVal UserId As Integer = 0)
    End Class

End Namespace

File: source/CPBase50/CPGroupBaseClass.cs
//========================================================================



//========================================================================

//
// documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
//


using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;

namespace Contensive.BaseClasses
{
	public abstract class CPGroupBaseClass
	{
		//
		// documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
		//
		public abstract void Add(string GroupName, string GroupCaption = ""); //Implements BaseClasses.CPGroupBaseClass.Add
        public abstract void AddUser(string GroupNameIdOrGuid);
        public abstract void AddUser(string GroupNameIdOrGuid, int UserId);
        public abstract void AddUser(string GroupNameIdOrGuid, int UserId, DateTime DateExpires);
        public abstract void Delete(string GroupNameIdOrGuid); //Implements BaseClasses.CPGroupBaseClass.Delete
		public abstract int GetId(string GroupNameIdOrGuid); //Implements BaseClasses.CPGroupBaseClass.GetId
		public abstract string GetName(string GroupNameIdOrGuid); //Implements BaseClasses.CPGroupBaseClass.GetName
		public abstract void RemoveUser(string GroupNameIdOrGuid, int UserId = 0); //Implements BaseClasses.CPGroupBaseClass.RemoveUser
	}

}

